src/ac_training_lab/A1-cam/device.py

```diff
@@ -0,0 +1,118 @@
+import paho.mqtt.client as mqtt
+import boto3
+import json
+from datetime import datetime
+import time
+
+from picamera2 import Picamera2
+from libcamera import controls, Transform
+
+from my_secrets import (
+    CAMERA_READ_ENDPOINT,
+    CAMERA_WRITE_ENDPOINT,
+    MQTT_PORT,
+    MQTT_HOST,
+    MQTT_PASSWORD,
+    MQTT_USERNAME,
+)
+
+import logging
+import sys
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+    handlers=[logging.StreamHandler(sys.stdout), logging.FileHandler("device.log")],
+)
+logger = logging.getLogger("device")
+
+
+def on_message(client, userdata, message):
```

Comment:
Any thoughts about queuing in this context? I.e., what will happen if it gets a request while it's still processing another (will it be ignored completely, will it sit in a queue, etc.?). This is more of a fringe case atm, but I wanted to double check what the expected behavior is

I think I designed https://ac-microcourses.readthedocs.io/en/latest/courses/hello-world/1.4-hardware-software-communication.html and the corresponding assignment microcontroller script to handle requests in a FIFO style, but the use-case is somewhat different and that uses MicroPython `mqtt_as` instead of Python `paho-mqtt` (see also [orchestrator nb](https://ac-microcourses.readthedocs.io/en/latest/courses/hello-world/1.4.1-onboard-led-temp.html)).